### PR TITLE
Use all Rails 7.1 defaults

### PR DESCRIPTION
**Pull request by chrisroos**
This PR upgrades the app to use the Rails 7.1 defaults instead of the 7.0 defaults.

I do that in a few stages:

- Run `rails app:update` using Rails 7.0.8 - it looks like this had been missed during a previous upgrade to Rails
- Run `rails app:update` using Rails 7.1.1 and accept all changes
- Revert changes made by `rails app:update` that we need
- Enable each new Rails 7.1 default in new_framework_defaults_7_1.rb and explain why I think each change is valid
- Use Rails 7.1 defaults in application.rb and remove new_framework_defaults_7_1.rb

There are quite a few commits but, apart from "Run rails app:update using Rails 7.1.1", most are small.

It's probably worth looking at the overall diff before tackling the commits individually as the resulting diff is smaller than some of the intermediate diffs.

### CODE REVIEW

File: config/application.rb
Comment (floehopper) [inline config/application.rb]: I assume it's safe to remove this because `load_defaults` is not set to `7.1`. Is that right? If so, it might be worth explaining in the commit note.

**Review by floehopper — APPROVED** [review]
Great work - very thorough! I've added several comments & questions inline. However, I don't think any of them should block merging, so I'm going to mark this as approved and leave it to your discretion whether there's anything that needs addressing.

**Comment by floehopper** [comment]
I've looked through my review again this morning and I'm not convinced any of my comments/questions are worth holding up merging this, so I'm going to merge it now.

File: config/initializers/new_framework_defaults_7_1.rb
Comment (chrisroos) [reply config/initializers/new_framework_defaults_7_1.rb]: As I've written above I relied on the tests to check for deprecation warnings.

File: config/application.rb
Comment (chrisroos) [reply config/application.rb]: Correct. I'm afraid it's too late to update the commit note now that it's in main.

**Comment by chrisroos** [comment]
Thanks for reviewing and merging this in my absence, @floehopper :+1:

**Pull request merged**
